lsp/server/src/server.ts

```diff
@@ -75,7 +91,7 @@ connection.onInitialized(() => {
         // Register for all configuration changes.
         connection.client.register(
             DidChangeConfigurationNotification.type,
-            undefined
+            { section: diagnosticsSettingSection }
```

Comment:
here we register to watch the change happens only under 'salesforce.salesforcedx-vscode-mobile.diagnostics' section.

Reply:
This is smart!